addons/product/models/product_catalog_mixin.py

```diff
@@ -40,7 +40,7 @@ def _get_product_catalog_domain(self):
         :returns: A list of tuples that represents a domain.
         :rtype: list
         """
-        return [('company_id', 'in', [self.company_id.id, False])]
+        return [('company_id', 'in', self.company_id.parent_path.split('/')[:-1] + [False])]
```

Comment:
Never had to use this term operator myself before, but this is where you can use `parent_of` for a cleaner solution:
```suggestion
        return ['|', ('company_id', '=', False), ('company_id', 'parent_of', self.company_id.id)]
```

Reply:
Had never seen this operator before, thank you! :slightly_smiling_face: